provider/pkg/gen/schema.go

```diff
@@ -433,6 +433,15 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 		}
 	}
 
+	// Apply auto-naming if
+	// a) There is an input property called name
+	// b) Its description doesn't contain a forward slash, which would indicate a custom format for the name.
+	//    We will tackle complex name auto-generation later on.
+	if nameProp, ok := inputProperties["name"]; ok && !strings.Contains(nameProp.Description, "/") {
+		requiredInputProperties.Delete("name")
+		resourceMeta.AutoName = true
+	}
```

Comment:
Just a thought - should we add a blurb to the description for the resource (or name property) to say that it supports autonaming?

Reply:
Or we could flip it to mark out resources that don't support autonaming yet. Just a compromise while we figure out how to improve overall coverage. I think getting some autonaming support through - even if partial is a win. I see a decent number of highly used resources already on the list...